### PR TITLE
feat: Select

### PR DESCRIPTION
**Pull request by ariansobczak-rst**
Created select component (https://react-select.com/) as a molecule.
Introduced styled **Select** and **SelectAsync** passing arguments as properties.

image

### CODE REVIEW

File: src/molecules/select/select-async.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    if (typeof onChange === 'function') onChange(selected)
+  }
+
+  return (
+    <ReactAsyncSelect
+      className={cssClass('Select')}
+      value={value}
+      styles={styles}
+      onChange={handleChange}
+      isClearable
```
Comment (dziraf) [inline src/molecules/select/select-async.tsx:27]: can you override `isClearable` via `selectProps`?

File: src/molecules/select/select.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        <FormGroup>
+          <Select value={value} onChange={(selected) => setValue(selected)} options={options} />
+        </FormGroup>
+      </StoryWrapper>
+      <StoryWrapper label="Select async example">
+        <FormGroup>
+          <SelectAsync
+            value={valueAsync}
+            onChange={(selected) => setValueAsync(selected)}
+            options={options}
```
Comment (dziraf) [inline src/molecules/select/select.stories.tsx:31]: shouldn't select async use `loadOptions` function instead of predefined options?

Comment (ariansobczak-rst) [reply src/molecules/select/select.stories.tsx]: Introduced simple Promise options
```ts
const loadOptions = async (inputValue: string): Promise<any[]> => {
    const options = [
      { value: 'chocolate', label: 'Chocolate' },
      { value: 'strawberry', label: 'Strawberry' },
      { value: 'vanilla', label: 'Vanilla' },
    ]

    setIsLoading(true)
    const found = filter(options, ({ value }) => value.includes(inputValue))
    setIsLoading(false)
    return Promise.resolve(found)
  }
  ```

File: src/molecules/select/select-async.tsx (hunk 1 above)
Comment (ariansobczak-rst) [reply src/molecules/select/select-async.tsx:27]: Yes, tried with example app and user resource

image
```
gender: {
  props: {
    isClearable: false,
  },
},
```
image

**Pull request merged**